src/PaymentMethods/Creditcard.php

```diff
@@ -39,7 +39,19 @@ public function getFormFields($generalFormFields): array
     public function hasPaymentFields(): bool
     {
         $componentsEnabled = $this->getProperty('mollie_components_enabled');
-        return $componentsEnabled ? $componentsEnabled === 'yes' : $this->defaultComponentsEnabled() === 'yes';
+        return $this->hasProperty('mollie_components_enabled') ? $componentsEnabled === 'yes' : $this->defaultComponentsEnabled() === 'yes';
+    }
+
+    /**
+     * Default values for the initial settings saved
+     *
+     * @return array
+     */
+    public function defaultSettings(): array
+    {
+        $settings = parent::defaultSettings();
+        $settings['mollie_components_enabled'] = $this->defaultComponentsEnabled();
```

Comment:
Is it still needed to add defaults in other gateways when `AbstractPaymentMethod` returns all defaults?